File: Jour22/jour22.c
#include<stdio.h>
#include<string.h>

int main()
{
	int i,k=0;
	char texte[50];
	printf("Ce programme test des palindromes.\n\n");
	do
	{
		printf("Donner un mot ou nombre de moins de 50 characteres : ");
		scanf("%s",&texte);
	}while(strlen(texte) > 49);

	for(i = 0; i < (int)(strlen(texte)/2); i++)
	{
		if(texte[i] == texte[strlen(texte)-(i+1)])
			k += 1;
	}
	
	if(k == (int)(strlen(texte)/2))
		printf("%s est un palindrome.",texte);
	else
		printf("%s n'est pas un palindrome.",texte);
	
	return 0;
}

File: Jour22/jour22.pas
program Palindrome;
var
   i,k              : integer;
   	texte          : string;
   
    
begin
	k := 0;
	writeln('Ce programme test des palindromes.');
	writeln;
	write('Donner un mot ou un nombre : ');
	read(texte);
    //Attention aucun control

    for i := 1 to trunc(length(texte)/2) do
    begin
        if texte[i] = texte[length(texte)-(i)+1] then
            k += 1;
    end;
	if k = trunc(length(texte)/2) then
	begin
		writeln(texte,' est un palindrome');
	end
	else
		writeln(texte,' n"est pas un palindrome');
end.